### DISCUSSION TITLE
CODEOWNERS REVIEWER

### DISCUSSION THREAD

**Discussion by TemitopeObasanmi** [community/community#138550] [New to GitHub]
I recently experience this issue having protected my branch with CODEOWNERS. There are meant to be 2 reviewers on the protected branch. For a test, i manually added someone who is not a CODEOWNER to review the PR. to my greatest surprise the person was able to review the PR.  This was not the result i was expecting, i expect that the individual i added manually should not have  access to review this code.

I would like to know if i did something wrong in my configuration of CODEOWNERS.

**Comment by spenserblack** [comment]
Anyone who can *read* a PR can review it. Even people who aren't maintainers. `CODEOWNERS` allows you to automatically *request* reviews from users or teams. For example:

```codeowners
*.js @js-expert
README.md @documentation-expert
```

This means that `@js-expert` will automatically get a review request for `*.js` files and `@documentation-expert` will get a review request for the `README.md`. It does *not* mean that `@js-expert` is forbidden from reviewing `README.md`.

You may want to look into [rulesets](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-rulesets/about-rulesets). Especially the ["Require a pull request before merging"](https://docs.github.com/en/repositories/configuring-branches-and-merges-in-your-repository/managing-rulesets/available-rules-for-rulesets#require-a-pull-request-before-merging) rule, which lets you specify "Require review from Code Owners".

**Reply by TemitopeObasanmi** [reply]
Thank you @spenserblack, i would like situation whereby only codeowners can review a code, is that possible? If it is how can i achieve this?

**Reply by spenserblack** [reply]
:thinking: Besides making your repository private, I don't know of a way to stop people from leaving reviews. If you really need this, you could possibly write a workflow that would automatically delete reviews. But I'm curious why you specifically want to block people from reviewing. Are you getting spam reviews?
